src/overlays/actors/ovl_Bg_Jya_Lift/z_bg_jya_lift.c

```diff
@@ -26,24 +34,132 @@ const ActorInit Bg_Jya_Lift_InitVars = {
+        if (this->unk168 >= 0x14) {
+
+            func_800800F8(globalCtx, 0xD66, -0x63, &this->dyna.actor, 0);
+            func_80899DBC(this);
+        }
+    } else {
+    }
+}
+
+void func_80899DBC(BgJyaLift* this) {
+    this->actionFunc = &BgJyaLiftMovementHandler;
+}
+
+void BgJyaLiftMovementHandler(BgJyaLift* this, GlobalContext* globalCtx) {
+    f32 distFromBottom;
+    {//Using a scope here made it match.  Not the best solution but it works.
+        f32 tempVelocity;
+        Math_SmoothScaleMaxMinF(&this->dyna.actor.velocity.y, 4.0f, 0.1f, 1.0f, 0.0f); // 0.10000000149011612f
+        tempVelocity = (this->dyna.actor.velocity.y < 0.2f) ? 0.2f : this->dyna.actor.velocity.y;
+        distFromBottom = Math_SmoothScaleMaxMinF(&this->dyna.actor.posRot.pos.y, 973.0f, 0.1f, tempVelocity, 0.2f); // 0.10000000149011612f 0.20000000298023224f
+    }
+    if ((this->dyna.actor.posRot.pos.y < 1440.0f) && (1440.0f <= this->dyna.actor.pos4.y)) {
+            func_8005B1A4(ACTIVE_CAM);
+        // Likely moves the camera down into the main room when the platform moves to y=1440;
+    }
+    if (fabsf(distFromBottom) < 0.001f) { //0.0010000000474974513f
+        BgJyaLiftSetFinalPosY(this);
+        Audio_PlayActorSound2(&this->dyna.actor, 0x287AU);
+        return;
+    }
+
+    func_8002F974((Actor*)this, 0x200CU);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Lift/BgJyaLift_Destroy.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Lift/func_80899D18.s")
+void BgJyaLiftSetFinalPosY(BgJyaLift* this) { //Sets final position after it falls to the bottom of its travel
+    this->actionFunc = 0;
+    this->dyna.actor.posRot.pos.y = 973.0f;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Lift/func_80899D38.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Lift/func_80899DBC.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Lift/func_80899DCC.s")
+void BgJyaLift_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgJyaLift* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Lift/func_80899EF0.s")
+    GlobalContext* globalCtx2 = globalCtx;
+    
+   // u8 u8Temp2; //Originally called phi_v0
+    if (this->actionFunc) {
+        (this->actionFunc)(this);
+    }
+    if ((((this->dyna.unk_160) & 4) != 0) && ((this->unk16B & 4) == 0)) {
+        func_8005A77C((Camera*)globalCtx2->cameraPtrs[0], 0x3F);
+    } else {
+        if (((this->dyna.unk_160) & 4) == 0 && ((this->unk16B & 4) != 0) && (globalCtx2->cameraPtrs[0]->setting == 0x3F)) {
+        func_8005A77C((Camera*)globalCtx2->cameraPtrs[0], 3);
+        }
+    }
+    this->unk16B = this->dyna.unk_160;
+    if ((globalCtx2->roomCtx.curRoom.num != 5) && (globalCtx2->roomCtx.curRoom.num != 25)) { 
+            Actor_Kill(thisx);
+       //Room 5 is the main room with the statue and room 25 is the room with the lift in it
```

Comment:
this comment is good to know. id say put it above the if probably, or above the actor_kill and indent it to keep it in line

Reply:
I'd also tweak the comment to read `Spirit Temple room 5 is the main room with the statue, while room 25 is directly above room 5`